src/Cli.Tests/EnvironmentTests.cs

```diff
@@ -0,0 +1,175 @@
+                }"
+        );
+        Assert.IsNotNull(resolvedJson);
+        Assert.IsTrue(JToken.DeepEquals(
+            JObject.Parse(
+                @"{
+                    ""envValue"": ""DEVELOPMENT"",
+                    ""hostingEnvValue"": ""PHOENIX_TEST""
+                }"),
+            JObject.Parse(resolvedJson)), "JSON resolved with environment variable correctly");
+
+        // Removing the .env file it will then use the value of system environment variable.
+
+    }
+
+    /// <summary>
+    /// Test to verify that no error is thrown if .env file is not present, and existing system variables is used.
+    /// </summary>
+    [TestMethod]
+    public void TestSystemEnvironmentVariableIsUsedInAbsenceOfEnvironmentFile()
+    {
+        Environment.SetEnvironmentVariable(TEST_ENV_VARIABLE, "TEST");
+        Assert.IsFalse(File.Exists(".env"));
+        DotNetEnv.Env.Load(); // No error is thrown
+        Assert.AreEqual("TEST", Environment.GetEnvironmentVariable(TEST_ENV_VARIABLE));
+    }
+
+    /// <summary>
+    /// Test to verify that if the environment variables are not resolved correctly, runtime engine will not start.
+    /// Here, in the first scenario, engine failed to start because the variable defined in the environment file
+    /// is typed incorrectly and do not match with the one present in the config.
+    /// </summary>
+    [DataRow("COMM_STRINX=test_connection_string", true, DisplayName = "Incorrect Variable name used in the environment file.")]
+    [DataRow("CONN_STRING=test_connection_string", false, DisplayName = "Correct Variable name used in the environment file.")]
+    [DataTestMethod]
+    public void TestFailureToStartWithUnresolvedJsonConfig(
+        string environmentFileContent,
+        bool isFailure
+    )
+    {
+        // Creating environment variable file
+        File.Create(".env").Close();
+        File.WriteAllText(".env", environmentFileContent);
+        if (File.Exists(TEST_RUNTIME_CONFIG_FILE))
+        {
+            File.Delete(TEST_RUNTIME_CONFIG_FILE);
+        }
+
+        string[] initArgs = { "init", "-c", TEST_RUNTIME_CONFIG_FILE, "--database-type", "mssql", "--connection-string", "@env('CONN_STRING')" };
+        Program.Main(initArgs);
+
+        // Trying to start the runtime engine
+        using Process process = ExecuteDabCommand(
+            "start",
+            $"-c {TEST_RUNTIME_CONFIG_FILE}"
+        );
+
+        string? output = process.StandardOutput.ReadToEnd();
+        Assert.IsNotNull(output);
+
+        if (isFailure)
+        {
+            // Failed to resolve the environment variables in the config.
+            Assert.IsFalse(output.Contains("Starting the runtime engine..."));
+            Assert.IsTrue(output.Contains("Error: Failed due to: Environmental Variable, CONN_STRING, not found."));
+            Assert.IsTrue(output.Contains("Error: Failed to start the engine."));
```

Comment:
I would argue that you should also assert that these two lines of console output are not present in the success scenario to directly validate your expectations, instead of relying on the presence of the string "starting the runtime engine"

Reply:
sure